modules/imgproc/src/imgwarp.cpp

```diff
@@ -2370,7 +2371,9 @@ static bool ocl_warpTransform(InputArray _src, OutputArray _dst, InputArray _M0,
     const char * const kernelName = op_type == OCL_OP_AFFINE ? "warpAffine" : "warpPerspective";
 
     int scalarcn = cn == 3 ? 4 : cn;
-    bool is32f = !dev.isAMD() && (interpolation == INTER_CUBIC || interpolation == INTER_LINEAR) && op_type == OCL_OP_AFFINE;
+    bool is32f = op_type == OCL_OP_AFFINE ?
+                 /* Affine*/ !dev.isAMD() && (interpolation == INTER_CUBIC || interpolation == INTER_LINEAR) :
```

Comment:
do we still have AMD-specific OpenCL kernel for warpAffine? If not, the condition should be updated.

Reply:
> do we still have AMD-specific OpenCL kernel for warpAffine?

There is conditional compilation with `AMD_DEVICE` in the cubic interpolation affine kernel:

https://github.com/opencv/opencv/blob/8e5dbc03fe0c8264e667de5bbae4d0ab04dcab6b/modules/imgproc/src/opencl/warp_affine.cl#L233-L245

https://github.com/opencv/opencv/blob/8e5dbc03fe0c8264e667de5bbae4d0ab04dcab6b/modules/imgproc/src/opencl/warp_affine.cl#L264-L283